### PR TITLE
Normalize e2e test snapshot stacktraces

### PR DESCRIPTION
**Pull request by kuceb**
in our e2e tests, lets replace the consecutive stack trace lines with one

example:
image

let me know if there's any lost coverage by doing this, but I cannot think of a compelling reason

this helps with firefox, since we're not always taking the same code paths

### User facing changes:
NA

### PR Tasks
- [x] add comment on regex

### CODE REVIEW

File: packages/server/test/support/helpers/e2e.coffee [hunk 1]
```diff
@@ -35,12 +35,12 @@ pathUpToProjectName = Fixtures.projectPath("")
 
 DEFAULT_BROWSERS = ['electron', 'chrome']
 
-stackTraceLinesRe = /^(\s+)at\s(.+)/gm
+stackTraceLinesRe = /^(\s*).*?(@|at).*\.(js|coffee|ts|html)(-\d+)?:\d+:\d+[\n\S\s]*?(\n\s*\n)/gm
```
Comment (bahmutov) [inline packages/server/test/support/helpers/e2e.coffee]: beautiful Regex

File: packages/server/test/support/helpers/e2e.coffee [hunk 2]
```diff
@@ -35,12 +35,12 @@ pathUpToProjectName = Fixtures.projectPath("")
...
 browserNameVersionRe = /(Browser\:\s+)(Custom |)(Electron|Chrome|Canary|Chromium|Firefox)(\s\d+)(\s\(\w+\))?(\s+)/
 availableBrowsersRe = /(Available browsers found are: )(.+)/g
 
 replaceStackTraceLines = (str) ->
```
Comment (bahmutov) [inline packages/server/test/support/helpers/e2e.coffee]: we don't have any unit tests for functions like this?

**Review by bahmutov — DISMISSED** [review]
to me this makes sense

File: packages/server/test/support/helpers/e2e.coffee (hunk 2 above)
Comment (flotwig) [reply packages/server/test/support/helpers/e2e.coffee]: no, we're really just rolling the dice and assuming we don't make any mistakes with these. the fact that we have tons of snapshots already provides *some* assurances, but there's no explicit testing of `e2e.coffee`.

**Review by flotwig — DISMISSED** [review]
I think the argument for keeping the lines is "you can tell if a change occurred"

But, since the error message is captured, we are still asserting that the proper error was thrown. If it was thrown from a slightly shallower or deeper stack, who cares?

I have made changes in the past that change the e2e stack trace and I didn't find it helpful to have them explicitly snapshotted, so I'm for this change :+1:

File: packages/server/test/support/helpers/e2e.coffee (hunk 2 above)
Comment (Bkucera) [reply packages/server/test/support/helpers/e2e.coffee]: I will add a comment on the regex so the intended behavior is at least written somewhere

**Review by flotwig — DISMISSED** [review]

**Comment by brian-mann** [comment]
Do not merge this PR, I am reviewing

**Comment by brian-mann** [comment]
I'm approving this PR but wanted to capture some feedback...

There are actually a number of situations where capturing the stack trace is an important user facing feature that's easy to mess up. We did a lot of modifications on the error objects - and there are many situations where we consume 3rd party user errors but then rethrow them again after building our own error. The stacks in these situations really matter.

There are also other scenarios where we conditionally enable Bluebird longStackTraces in `open` mode vs `run` mode which will change how the stacks appear. That information will be lost by this PR - but with the addition of firefox, it can't be helped without us doing a lot of extra work.

HOWEVER- i think in the future we could introduce something into the snapshot subsystem that does something like..

```
An error message
    [chrome: 5 stack lines]
    [firefox: 8 stack lines]
```

This would capture the stacks independently for each browser but combine them into a single snapshot. We'd need to update `snap-shot-it` to be able to handle this scenario by allowing us to first read off existing snapshots prior to comparing them.

That work isn't worth doing right now, but at least it is something we could do in the future.

Additionally the work that @chrisbreiding is doing on the Error Improvements will likely also cover some of the lost coverage we're losing by removing these stack traces.

**Review by brian-mann — APPROVED** [review]

**Pull request merged**